scipy/integrate/_ivp/ivp.py

```diff
@@ -511,6 +511,15 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
         # Wrap the user's fun (and jac, if given) in lambdas to hide the
         # additional parameters.  Pass in the original fun as a keyword
         # argument to keep it in the scope of the lambda.
+        try:
```

Comment:
Instead I would have just used `args = np.atleast_1d(args)`. Nothing else would be needed (or we could also check the resulting ndim/shape).

As for the current code, you should use f-strings for concatenation and also `raise TypeError(...) from exc`.

Reply:
String concat is fixed (as I understood it)?

I use suggestion_tuple variable, because if I put a multi-line string in raise TypeError, it will be printed ugly:

```
File ".../python3.7/site-packages/scipy/integrate/_ivp/ivp.py", line 539, in solve_ivp
    "Supplied 'args' cannot be unpacked. If you have " ( <-THIS LINE just confusing)
TypeError: Supplied 'args' cannot be unpacked. If you have one argument, try coma in the end of tuple. Ex. args=(2,)
```

Thath's how it looks now:

```
File ".../python3.7/site-packages/scipy/integrate/_ivp/ivp.py", line 538, in solve_ivp
    raise TypeError(suggestion_tuple)
TypeError: Supplied 'args' cannot be unpacked. If you have one argument, try coma in the end of tuple. Ex. args=(2,)
```

Regarding `raise TypeError(...) from exc`, according to [Python Doc](https://docs.python.org/3/tutorial/errors.html#exception-chaining), 

> Exception chaining happens automatically when an exception is raised inside an except or finally section. Exception chaining can be disabled by using from None idiom:

Reply:
Yes you fixed the chaining. Just a style note, we prefer now to have the space at the beginning of the new line.

As for the exception. Yes the chain will be present, but the message is clearer with using `raise ... from exc`. Without it you get:

`During handling of the above exception, another exception occurred:`

and with `from`:

`The above exception was the direct cause of the following exception:`

Reply:
@tupui Thanks, styling fixed again. (Wasn't able to find styling guide, so I used one from pandas)

With error chaining - you win, fixed also.

Reply:
True, a styling guide is in preparation #13955